### PR TITLE
Add --certificate options to creds and measure

### PR DESCRIPTION
**Pull request by daandemeyer**
Let's allow loading the public key to use directly from a certificate instead of requiring the user to extract the public key themselves.

### CODE REVIEW

File: man/systemd-creds.xml [hunk 1]
```diff
@@ -353,6 +353,14 @@
         be possible without losing access to all previously encrypted secrets.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--tpm2-certificate=</option><arg>PATH</arg></term>
+
+        <listitem><para>Takes a path to a PEM encoded X509 certificate. If specified,
+        <command>systemd-creds</command> will bind the encryption to the certificate's public key. This
+        option cannot be used together with <option>--tpm2-public-key</option>.</para></listitem>
```
Comment (poettering) [inline man/systemd-creds.xml:361]: suffix `--tpm2-public-key` with `=` please

hmm, wouldn't it be nicer to just make the existing `--tpm2-public-key=` key switch to also read a certificate if one is passed? Sounds simpler to me, then having two distinct options.

i.e. try to parse as cert and if it doesn't, parse as public key? there should not be any ambiguity possible

File: src/creds/creds.c [hunk 2]
```diff
@@ -492,14 +494,36 @@ static int verb_encrypt(int argc, char **argv, void *userdata) {
...
+                else
+                        r = tpm2_load_pcr_public_key(arg_tpm2_public_key, &pubkey, &pubkey_size);
+                if (r < 0) {
+                        if (arg_tpm2_public_key || arg_tpm2_public_key || r != -ENOENT || !sd_id128_in_set(arg_with_key, _CRED_AUTO, _CRED_AUTO_INITRD))
+                                return log_error_errno(r, "Failed read TPM PCR public key: %m");
+
+                        log_debug_errno(r, "Failed to read TPM2 PCR public key, proceeding without: %m");
+                }
+
+                log_info("pubkey: %s", (char *) pubkey);
```
Comment (poettering) [inline src/creds/creds.c:516]: left-over debugging?

**Comment by poettering** [comment]
conceptually sounds great, but I think we should just reuse the existing switch and just allow it to accept both pubkeys and certs the same way, doing the right thing automatically.

moreover, we should teach systemd-measure to generate the pubkey automatically from the privkey, so that you only have to specify the latter.

**Comment by DaanDeMeyer** [comment]
It seems measure already supports deriving the public key from the private key. That's actually much preferable so I'll close this PR

**Pull request closed without merging**